x/evm/types/state_transition.go

```diff
@@ -246,25 +268,28 @@ func (st StateTransition) TransitionDb(ctx sdk.Context, config ChainConfig) (*Ex
 		"executed EVM state transition; sender address %s; %s", st.Sender.String(), recipientLog,
 	)
 
+	// Consume gas from evm execution
+	// Out of gas check does not need to be done here since it is done within the EVM execution
+	ctx.WithGasMeter(currentGasMeter).GasMeter().ConsumeGas(gasConsumed, "EVM execution consumption")
```

Comment:
since you add only `gasConsumed` to the gas meter, i don't think you need to add refunded gas back to the block gas meter since it isn't consumed in the first place

Reply:
hmm do you think you can verify for me that that is what we're doing?
aka, is block gas meter == refund counter ?
because from what ive looked at so far, theyre different things. but i may be wrong here!